mperf.go

```diff
@@ -0,0 +1,58 @@
+//
+// MinIO Object Storage (c) 2022 MinIO, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package madmin
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+// MperfNodeResult - stats from each server
+type MperfNodeResult struct {
+	Endpoint string `json:"endpoint"`
+	TX       uint64 `json:"tx"`
+	RX       uint64 `json:"rx"`
+	Error    string `json:"error,omitempty"`
+}
+
+// MperfResult - aggregate results from all servers
+type MperfResult struct {
+	NodeResults []MperfNodeResult `json:"nodeResults"`
+}
+
+// Mperf - perform mperf on the MinIO servers
+func (adm *AdminClient) Mperf(ctx context.Context, duration time.Duration) (result MperfResult, err error) {
+	queryVals := make(url.Values)
+	queryVals.Set("duration", duration.String())
+
+	resp, err := adm.executeMethod(ctx,
+		http.MethodPost, requestData{
+			relPath:     adminAPIPrefix + "/mperf",
+			queryValues: queryVals,
+		})
+	if err != nil {
+		return result, err
+	}
+	if resp.StatusCode != http.StatusOK {
+		return result, httpRespToErrorResponse(resp)
+	}
+	err = json.NewDecoder(resp.Body).Decode(&result)
+	return result, err
+}
```

Comment:
I think its better to do it like an option for `speedtest` @krishnasrinivas not as a new API

Reply:
Doing it as one API will make the API complicated @harshavardhana for ex. the response structure is different for mperf, dperf and object tests

Reply:
No it needs to be a new SpeedTestV2 API that replies back a data structure like this 

```
type SpeedtestResult struct {
         NetworkInfo
         DiskInfo
         ObjectLayerInfo
}
```

This provides a more consistent view from `mc` - separate APIs are not needed IMO.  This is in line with how subnet health does things.